java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/menu/handlers/UpdateCategoryHandler.java

```diff
@@ -35,16 +35,33 @@ public UpdateCategoryHandler(MenuRepository repository,
     return Optional.of(command.getCategoryId());
   }
 
+  /**
+   * if the request is to update the name and description of a category
+   * then checks if there is a category with the same name exists.
+   * If there is a category with the same name but the update is trying to update the description
+   * then allow that
+   * else throw a category already exists exception
+   * if a category with the same name doesn't exits then update the requested category.
+   *
+   * @param menu    menu
+   * @param command update category request
+   * @return category
+   */
   Category updateCategory(Menu menu, UpdateCategoryCommand command) {
     Category category = getCategory(menu, command);
+    menu.getCategories().forEach(t -> {
+      if (t.getName().equalsIgnoreCase(command.getName())) {
+        if (t.getId().equalsIgnoreCase(command.getCategoryId().toString())) {
+          category.setDescription(command.getDescription());
+        } else {
+          throw new CategoryAlreadyExistsException(command, command.getName());
+        }
+      } else {
+        category.setDescription(command.getDescription());
+        category.setName(command.getName());
```

Comment:
The description is set regardless of this check `t.getId().equalsIgnoreCase(command.getCategoryId().toString()`, is it expected?